### ISSUE TITLE
Include 'False' as possible false value for boolean field

### ISSUE THREAD

**Issue by abendebury** [wtforms/wtforms#290]
https://github.com/wtforms/wtforms/blob/b94260329964844d66c426072c649e64b6924a7c/wtforms/fields/core.py#L696

When submitting a form in a unit test using the flask client, if you do this:

```
client.post("/", data={"field": False})
```

The `False` value will become the string `"False"`, which does not register as `False` in a BooleanFIeld.

This would be fixed by adding `"False"` to line 696.

**Comment by davidism** [comment]
Seems fine, if you could make a PR with a test, that would really help me out.

**Comment by davidism** [comment]
Reversing my stance on this. This is currently for parsing HTML form data. HTML forms don't send a value for unchecked inputs. When testing, you should be like a client and not send unchecked values either. I'm not even clear why we accept `'false'` right now.

Even if we refactored to support JSON data, a boolean would deserialize to `True` or `False` before being passed to the form, so checking for the string still doesn't make sense.

If you really want to accept other values, you can already override it by passing in `false_values` or using a subclass that sets the `false_values` class attribute.

**Comment by emin63** [comment]
Well, you're the boss I guess. I'm not sure why you would reject this pull request that helps save people from subtle bugs. I agree with you that HTML forms are stupid in how they handle unchecked inputs and I guess it's consistent to aim for similar behavior in wtforms.

I can pretty much guarantee you that people who want to use wtforms to build a system which has both forms for human interaction and also hits those same forms in an automated way so they don't have to build a separate REST API is going to get hit by this issue.

Anyway, thanks for your great work on wtforms and for listening to my rant. :)

Best wishes.

**Comment by victorBuzdugan** [comment]
Maybe you changed your mind about this because it's really misleading and introduces subtle bugs when unit testing.
Right now the default false values as `simple.py` line 33:
```false_values = (False, "false", "")```
Having `False` in there is misleading because one should assume that posting, in a unit test, data from a user object will get in the form.data route the boolean `False` but instead gets `True`, yes because what you are actually posting is the string `"False"` which because isn't in `false_values` and is not `None` evaluates to `True`.
```python
data = {
    "csrf_token": g.csrf_token,
    "name": user.name,
    "details": user.details,
    "email": user.email,
    # BooleanField
    "check_inv": user. check_inv,
    # BooleanField
    "admin": user.admin,
    # BooleanField
    "in_use": user.in_use,
    # BooleanField
    "sat_group": user.sat_group,
    "clean_order": "0",
    "submit": True,
}
response = client.post(
    url_for("users.edit_user", username=user.name),
    data=data)
```
I understand that this is a HTML form limitation and that we should send unchecked values also, just like HTML forms, but when testing you don't really know what this value will be and this introduces extra checks or extra code like `str(user. check_inv).lower()` to match `"false"` value (by the way just like you said I don't see the point of `"false"` there).

Overriding `false_values` on each BooleanField also introduces extra code on each field, and a thing you need to remember to pass each time you are making a BooleanField.

To summarise:
- having `False` in `false_values` is misleading
- having `"false"` in `false_values` seems useless 
- having `"False"` in `false_values` will make BooleanField **behave as expected** and in my opinion more pythonic (maybe also `"0"` and `0`)

Please, if you have the time, take another look at this.
